### DISCUSSION TITLE
Is it possible to automate (via CLI) the connection between GitHub and Vercel?

### DISCUSSION THREAD

**Discussion by jeanfrancoislarente** [vercel/vercel#8240] [Help] [answered]
We're trying to automate some deployments via a portal where users can authenticate to GitHub (using OAuth and the correct flow).

Is there a way using the Vercel API / CLI to programmatically / automatically connect a GitHub repo to a Vercel account?

**Comment by amyegan** [comment, marked as answer]
Hi @jeanfrancoislarente. If the [Vercel for Git](https://vercel.com/docs/concepts/git/vercel-for-github#) integration will not work for your use case, you can Vercel CLI with a GitHub Action to automate Project deployment. More details here https://vercel.com/support/articles/how-can-i-use-github-actions-with-vercel

**Reply by jeanfrancoislarente** [reply]
Thanks for the suggestion. 

Unfortunately what we're trying to do is automate the connectivity between GitHub and Vercel for a 3rd party (so to speak) to automate the deployments on their behalf.

We have a custom automation process that doesn't use GitHub Actions so that won't be an option for us.

Cheers!
